memory_diff/diff.py

```diff
@@ -9,33 +13,88 @@ def __init__(self,file_new: str, file_old:str, file_diff:str) -> None:
         
-                                                                              # Starting the research of the differences between two files
-        for string_new in self.list_file_new:
-            exist: bool = False
-            i:int = 0
-            while i<len(self.list_file_old) and exist == False:
-                if string_new == self.list_file_old[i] :
+        new_list_tag_tu = Bs_data_new.find_all('tu')                          # Saving all tag <tu> (with all their descendants) into a list
+        old_list_tag_tu = Bs_data_old.find_all('tu')
+
+
+            ## For each tag <tu> of the new list, create an object TranslatedUnit (using the self.build_object_tu()) and compare that object with all the TranslatedUnit
+            ## of the old list, and, if there is not a match, add the object into another list that contains only the added/modified elements
+        list_block_diff=[]
+        for tu_tag in new_list_tag_tu:                                        # Begins loop
+            tu_object_new = TranslatedUnit(tu_tag)                            # Creates the object to compare
+            exist:bool = False 
+            for i in range(len(old_list_tag_tu)):                             # Does the comparing
+                if exist:
+                    break
+                tu_object_old = TranslatedUnit (old_list_tag_tu[i])
+                if tu_object_new == tu_object_old:
                     exist = True
-                i=i+1
-            if(not exist):
-                self.list_file_diff.append(string_new)
-
-                                                                            # Saving the differences into a third file called diff_file
-        file_object_diff = open(self.file_diff, 'w')
-        file_object_diff.writelines(self.list_file_diff)
-        file_object_diff.close()                                             
\ No newline at end of file
+            if not exist:
+                list_block_diff.append(tu_object_new)
+        
+        self.saving_diff(list_block_diff)                                # Invoke saving_diff method to save the diff block into a file
+
+    
+    
+    # Saving the differences into a third file called diff_file
+    def saving_diff(self,list_block_diff):
+        Bs_data_new = BeautifulSoup(self.opened_file_new, "xml")
+        tag_header_of_new_file = Bs_data_new.find('header')
+        root = ET.Element('tmx')                                               # Root Creation
+        root.set('version','1.4')                                              # Setting root's attributes
+        
+        header = ET.SubElement(root,'header')                                  # Header Creation as root's subelement 
+        header.set('creationtool',tag_header_of_new_file['creationtool'])      # Setting header's attributes
+        header.set('creationtoolversion',tag_header_of_new_file['creationtoolversion'])
```

Comment:
io splitterei questa logica in funzioni diverse, del tipo:
- `build_header()` --> per settare i parametri nell'header
- `build_tu()` --> per settare i parametri nella tu

per snellire un pò la funzione saving_diff